parser/src/parser.rs

```diff
@@ -204,19 +204,23 @@ impl<'a> Parser<'a> {
 
     let contents_end = self.current_location();
     let mut after_invalid = None;
+    let mut opening_quote_loc = None;
 
     loop {
       match self.peek() {
-        Some((_, '}')) => {
+        Some((loc, '|')) => {
           self.next();
-          if let Some(invalid_end) = after_invalid {
-            self.report(Diagnostic::PlaceholderInvalidContents {
-              span: Span::new(contents_end..invalid_end),
-            });
+          if opening_quote_loc.is_none() {
```

Comment:
Can we call `self.parse_quoted()` instead of `self.next()`+this logic? Like we already do for escapes